### ISSUE TITLE
False positive for `Style/HashAsLastArrayItem` when the array is implicit

### ISSUE THREAD

**Issue by dvandersluis** [rubocop-hq/rubocop#8740]
`Style/HashAsLastArrayItem` detects and corrects hashes in an implicit array (such as when passing multiple values to a setter), resulting in a syntax error.

--------

## Expected behavior

The following is a common line of code in Rails projects:

```ruby
config.cache_store = :redis_cache_store, { url: ENV['REDIS_URL']  }
```

It should not register an offense, as the braces are necessary syntax.

## Actual behavior

Rubocop registers an offense:

```console
bundle exec rubocop bug.rb -A --only Style/HashAsLastArrayItem
bug.rb:1:42: C: Style/HashAsLastArrayItem: Omit the braces around the hash.
config.cache_store = :redis_cache_store, { url: ENV['REDIS_URL'] }
                                         ^^^^^^^^^^^^^^^^^^^^^^^^^
```

And corrects to:

```ruby
config.cache_store = :redis_cache_store,  url: ENV['REDIS_URL'] 
```

Which is a syntax error:

```console
$ ruby bug.rb
bug.rb:1: syntax error, unexpected tLABEL
...ore = :redis_cache_store,  url: ENV['REDIS_URL']
```

## RuboCop version

```
$ rubocop -V
0.91.0 (using Parser 2.7.1.4, rubocop-ast 0.4.0, running on ruby 2.6.2 x86_64-darwin18)
```
